app/hooks/StrapiAPI.ts

```diff
@@ -1,32 +1,280 @@
+/**
+  NOTE @rosschapman: Developers may be tempted to auto-generate response types as described in the strapi docs using cli
+  commands from the `@strapi/strapi` module, but I've noticed the output is funky for use in a client application in its
+  raw form. For example, string fields from the recommended module are typed as `Attribute.String` but this type isn't compatible
+  with `string` 🤦. Strapi offers this cumbersome approach to sync types, but also caveats this is not an official
+  solution: https://strapi.io/blog/improve-your-frontend-experience-with-strapi-types-and-type-script. Even so, I still
+  don't trust the generated types in view of the above example.
+*/
+
 import useSWR from "swr";
 import fetcher from "utils/fetcher";
-import { type Footer, type StrapiResponse } from "models/Strapi";
 import config from "../config";
 
 interface SWRHookResult<T> {
-  data: T | null;
+  data: StrapiApi.StrapiDataResponse<T>;
   error?: Error;
   isLoading: boolean;
 }
 
 function useStrapiHook<T>(path: string): SWRHookResult<T> {
   const dataFetcher = () =>
-    fetcher<StrapiResponse<T>>(`${config.STRAPI_API_URL}/api/${path}`, {
-      Authorization: `Bearer ${config.STRAPI_API_TOKEN}`,
-    });
+    fetcher<StrapiApi.StrapiResponse<T>>(
+      `${config.STRAPI_API_URL}/api/${path}`,
+      {
+        Authorization: `Bearer ${config.STRAPI_API_TOKEN}`,
+      }
+    );
 
-  const { data, error, isLoading } = useSWR<StrapiResponse<T>>(
+  const { data, error, isLoading } = useSWR<StrapiApi.StrapiResponse<T>>(
     `/api/${path}`,
     dataFetcher
   );
 
   return {
-    data: data?.data ? data.data.attributes : null,
+    data: data?.data ? data.data : null,
     error,
     isLoading,
   };
 }
 
 export function useFooterData() {
-  return useStrapiHook<Footer>("footer?populate[links][populate]=*");
+  return useStrapiHook<StrapiApi.FooterResponse>(
+    "footer?populate[links][populate]=*"
+  );
+}
+
+export function useHomepageData() {
+  return useStrapiHook<StrapiApi.HomepageResponse>(
+    "home-page?populate[hero][populate]=*&populate[category_section][populate]=*&populate[opportunity_section][populate]=*&populate[opportunities][populate][image][populate]=*&populate[opportunities][populate][address]=*&populate[opportunities][populate][links][populate]=*&populate[opportunities][populate][opportunity_categories]=*&populate[event_section][populate]=*&populate[events][populate][date]=*&populate[events][populate][address]=*&populate[events][populate][links]=*&populate[events][populate][image][populate]=*&populate[events][populate][event_categories]=*&populate[two_column_content_blocks][populate][link]=*&populate[two_column_content_blocks][populate][media][populate]=*"
+  );
```

Comment:
+1! Is there a way to just pull everything from the homepage or does this cause type issues? I guess this is the trade-off with not using graphQL.

Reply:
@adriencyberspace , if you specify `home-page?populate=*` it will pull **almost** everything from the homepage, but relationships, components and dynamic zones will not be fully populated, unfortunately.